tdh/resource_user.go

```diff
@@ -348,19 +408,60 @@ func (r *userResource) Read(ctx context.Context, req resource.ReadRequest, resp
 		return
 	}
 
-	// Get refreshed cluster value from TDH
-	user, err := r.client.CustomerMetadata.GetUser(state.ID.ValueString())
-	if err != nil {
-		resp.Diagnostics.AddError(
-			"Reading TDH user",
-			"Could not read TDH user ID "+state.ID.ValueString()+": "+err.Error(),
-		)
-		return
-	}
+	query := &customer_metadata.UsersQuery{}
+	if r.client.Root.IsSre {
+		user, err := r.client.CustomerMetadata.GetUsers(query, r.client.Root.IsSre)
+		if err != nil {
+			resp.Diagnostics.AddError(
+				"Reading TDH user",
+				"Could not read TDH user ID "+state.ID.ValueString()+": "+err.Error(),
+			)
+			return
+		}
+		if user.Page.TotalPages > 1 {
+			for i := 1; i <= user.Page.TotalPages; i++ {
+				query.PageQuery.Index = i - 1
+				page, err := r.client.CustomerMetadata.GetUsers(query, r.client.Root.IsSre)
+				if err != nil {
+					resp.Diagnostics.AddError(
+						"Unable to Read User",
+						err.Error(),
+					)
+					return
+				}
+
+				for _, dto := range *page.Get() {
+					if dto.Id == state.ID.ValueString() {
+						if saveFromUserResponse(&ctx, &resp.Diagnostics, &state, &dto, false) != 0 {
+							return
```

Comment:
last param of this method should have been `true`, no? ... anyway put that isSre check thing inside, so it can not be missed,